buy.ts

```diff
@@ -398,9 +489,11 @@ function loadSnipeList() {
 }
 
 function shouldBuy(key: string): boolean {
-  return USE_SNIPE_LIST ? snipeList.includes(key) : true;
+  logger.info(processingToken, 'Is processing token buy')
+  return USE_SNIPE_LIST ? snipeList.includes(key) : ONE_TOKEN_AT_A_TIME ? !processingToken : true
```

Comment:
I don't think processingToken check is going to work all the time. 
```
        const _ = processRaydiumPool(updatedAccountInfo.accountId, poolState);
```
is not awaited and there is no check for processingToken, we could enter buy operation multiple times.
For this feature, I would use: https://www.npmjs.com/package/async-mutex and fix the logic.

Reply:
I'm taking a look at the project, I hope I'm not talking nonsense, but concurrency code could be problematic, by my point of view is  better to avoid concurrency and use a sequential logic with a queue.  putting the processToken as callback (bind) in a queue, and use as scheduler that sequentially dequeue and execute processToken sequentially. The scheduler could be a main loop that perform regular job  and log stats regularly.